core/lib/zksync_core/src/eth_watch/event_processors/set_chain_id.rs

```diff
@@ -0,0 +1,78 @@
+use zksync_contracts::SET_CHAIN_ID_EVENT;
+use zksync_dal::StorageProcessor;
+use zksync_types::{protocol_version::decode_set_chain_id_event, web3::types::Log, Address, H256};
+
+use crate::eth_watch::{
+    client::{Error, EthClient},
+    event_processors::EventProcessor,
+    metrics::{PollStage, METRICS},
+};
+
+/// Responsible for saving `setChainId` upgrade transactions to the database.
+#[derive(Debug)]
+pub struct SetChainIDEventProcessor {
```

Comment:
I like this idea, this would simplify a lot of things. But don't we need to run genesis before deploying the contracts? I think it logs some config variables used in contracts later, like the genesis batch commitment etc